docs/src/components/copy-button.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { CheckIcon, CopyIcon } from "@radix-ui/react-icons";
+
+import { cn } from "@/lib/utils";
+import { Button } from "@/components/ui/button";
+
+interface CopyButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
+  value: string;
+  src?: string;
+}
+
+async function copyToClipboard(value: string) {
+  navigator.clipboard.writeText(value);
+}
+
+export function CopyButton({
+  value,
+  className,
+  src,
+  ...props
+}: CopyButtonProps) {
+  const [hasCopied, setHasCopied] = React.useState(false);
+
+  React.useEffect(() => {
+    setTimeout(() => {
+      setHasCopied(false);
+    }, 2000);
+  }, [hasCopied]);
```

Comment:
This dependency is fragile because the effect doesn't actually rely on it. A better approach would be to include it in the code for useEffect so that it's part of the effect's life-cycle. you can, for example, check that it is in fact true before setting the timeout.